tools/build_executable.py

```diff
@@ -0,0 +1,105 @@
+    elif sys.platform == "darwin":
+        cmd_separator = ";"
+        run_uat_script = os.path.join(args.unreal_engine_dir, "Engine", "Build", "BatchFiles", "RunUAT.sh")
+        target_platform = "Mac"
+    elif sys.plaform == "linux":
+        cmd_separator = ";"
+        run_uat_script = os.path.join(args.unreal_engine_dir, "Engine", "Build", "BatchFiles", "RunUAT.sh")
+        target_platform = "Linux"
+    else:
+        assert False
+
+    # clone repo along with submodules  
+    cmd = ["git", "clone", "--recurse-submodules", "https://github.com/isl-org/spear", repo_dir]
+    print(f"[SPEAR | build.py] Executing: {' '.join(cmd)}")
+    cmd_result = subprocess.run(cmd)
+    assert cmd_result.returncode == 0
+
+    # change working directory
+    cwd = os.getcwd()
+    print(f"[SPEAR | build.py] Changing working directory to {os.path.join(repo_dir, 'tools')}")
+    os.chdir(os.path.join(repo_dir, "tools"))
+
+    if args.commit_id:
+        cmd = ["git", "reset", "--hard", args.commit_id]
+        print(f"[SPEAR | build.py] Executing: {' '.join(cmd)}")
+        cmd_result = subprocess.run(cmd)
+        assert cmd_result.returncode == 0
+
+    # build thirdparty libs
+    cmd = ["python", "build_third_party_libs.py", "--num_parallel_jobs", f"{args.num_parallel_jobs}"]
+    print(f"[SPEAR | build.py] Executing: {' '.join(cmd)}")
+    cmd_result = subprocess.run(cmd, shell=True, executable=args.shell_bin)
+    assert cmd_result.returncode == 0
+
+    # create symbolic links
+    cmd = ["conda", "activate", args.conda_env + cmd_separator, "python", "create_symbolic_links.py"]
+    print(f"[SPEAR | build.py] Executing: {' '.join(cmd)}")
+    cmd_result = subprocess.run(cmd, shell=True, executable=args.shell_bin)
+    assert cmd_result.returncode == 0
+
+    # generate config file
+    cmd = ["conda", "activate", args.conda_env + cmd_separator, "python", "generate_config.py"]
+    print(f"[SPEAR | build.py] Executing: {' '.join(cmd)}")
+    cmd_result = subprocess.run(cmd, shell=True, executable=args.shell_bin)
+    assert cmd_result.returncode == 0
+    
+    archive_dir = os.path.join(args.output_dir, f"SpearSim-{target_platform}-{args.config_mode}")
+    cmd = [
+        run_uat_script,
+        "BuildCookRun",
+        "-project=" + os.path.join(repo_dir, "cpp", "unreal_projects", "SpearSim", "SpearSim.uproject"), 
+        "-build",
+        "-cook",
+        "-stage",
+        "-package",
+        "-archive",
+        "-pak",
+        "-targetplatform=" + target_platform,
+        "-target=SpearSim",
+        "-archivedirectory=" + archive_dir,
+        "-clientconfig=" + args.config_mode
+    ]
+    print(f"[SPEAR | build.py] Executing: {' '.join(cmd)}")
+    cmd_result = subprocess.run(cmd)
+    assert cmd_result.returncode == 0
+
+    # change working directory
+    print(f"[SPEAR | build.py] Changing working directory to {cwd}")
+    os.chdir(cwd)
+
+    print(f"[SPEAR | build.py] Successfully built SpearSim at {archive_dir}.")
```

Comment:
- do one last print statement after this to match the rest of our code: "[SPEAR | build_executable.py] Done."
- do the same thing in your other PRs

Reply:
Done